controllers/api/userRoutes.js

```diff
@@ -1,61 +1,222 @@
 const router = require('express').Router();
-const { User } = require('../../models');
+const { User, Event, Task } = require('../../models');
+const withAuth = require('../../utils/auth');
 
-router.post('/', async (req, res) => {
-  try {
-    const userData = await User.create(req.body);
+// GET /api/users
+router.get('/', (req, res) => {
+  // Access our User model and run .findAll() method
+  User.findAll({
+      attributes: { exclude: ['email'] }
+  })
+    .then(dbUserData => res.json(dbUserData))
+    .catch(err => {
+      console.log(err);
+      res.status(500).json(err);
+    });
+});
 
-    req.session.save(() => {
-      req.session.user_id = userData.id;
-      req.session.logged_in = true;
+// GET /api/users/1
+router.get('/:id', (req, res) => {
+  User.findOne({
+      attributes: { exclude: ['email']},
+      where: {
+        id: req.params.id
+      },
+      include: [
+          {
+            model: Event,
+            attributes: ['id', 'name', 'event_date', 'event_address', 'event_owner']
+          },
+          {
+              model: Task,
+              attributes: ['id', 'name', 'event_id', 'volunteer'],
+              include: {
+                model: Event,
+                attributes: ['name']
+              }
+          }
+        ]
 
-      res.status(200).json(userData);
+  })
+    .then(dbUserData => {
+      if (!dbUserData) {
+        res.status(404).json({ message: 'No user found with this id' });
+        return;
+      }
+      res.json(dbUserData);
+    })
+    .catch(err => {
+      console.log(err);
+      res.status(500).json(err);
     });
-  } catch (err) {
-    res.status(400).json(err);
-  }
 });
 
-router.post('/login', async (req, res) => {
-  try {
-    const userData = await User.findOne({ where: { email: req.body.email } });
+// POST /api/users
+router.post('/', (req, res) => {
+  User.create({
+    first_name: req.body.first_name,
+    last_name: req.body.last_name,
+    email: req.body.email,
+    password: req.body.password
+  })
+  .then(dbUserData => {
+    req.session.save(() => {
+      req.session.user_id = dbUserData.id;
+      req.session.first_name = dbUserData.first_name;
+      req.session.last_name = dbUserData.last_name;
+      req.session.password = dbUserData.password;
```

Comment:
I don't think we want this info in the session data.  We can just pull it from the database if we need it.  I think this is also risky from a security standpoint.